dev

shestooy opened the pull request.

Commit:
initial commit

Commit:
change directory README.MD

Commit:
change directory README.MD

shestooy commented:
Привет, я тут немного самодеятельности допустил
Чтобы Ваше время сэкономить я написал для вас README.MD
Посмотрите его в запушенных файлах

Commit:
Merge remote-tracking branch 'origin/DEV' into DEV

ivannizh commented on internal/storage/storage.go:
```diff
@@ -0,0 +1,141 @@
+}
+
+func (s *ParcelStore) Add(p Parcel) (int64, error) {
+	res, err := s.db.Exec("INSERT INTO parcel (client,status,address,created_at) VALUES (:c, :s, :a, :cr)",
```
Если нет желания полноценно называть переменные, то можно использовать неименованные параметры
```suggestion
	res, err := s.db.Exec("INSERT INTO parcel (client,status,address,created_at) VALUES (?, ?, ?, ?)", p.Client, p.Status, p.Address, p.CreatedAt)
```

ivannizh commented on internal/storage/storage.go:
```diff
@@ -0,0 +1,141 @@
+		}
+
+		res = append(res, p)
+	}
```
После цикла нужно проверить курсор на наличие ошибок

http://go-database-sql.org/retrieving.html

ivannizh commented on internal/storage/storage.go:
```diff
@@ -0,0 +1,141 @@
+}
+
+func (s *ParcelStore) SetAddress(number int64, address string) error {
+	row := s.db.QueryRow("SELECT status FROM parcel WHERE number = :n",
```
В этой и следующей функции нужно упростить логику и делать все в одном запросе.

Нужно убрать первый запрос и добавить в `UPDATE`(`DELETE`) еще одно условие в секцию `WHERE`

ivannizh commented on internal/storage/storage_test.go:
```diff
@@ -0,0 +1,150 @@
+
+	// check
+	for _, parcel := range storedParcels {
+		require.Equal(t, parcel, parcelMap[parcel.Number])
```
Это хорошая проверка, но по ТЗ перед этим нужно проверить еще наличие такого айди в мапе. Это можно сделать с помощью средств мапы (когда она возвращает 2 значения)

ivannizh reviewed (commented):
Привет! Хорошая работа! Осталось исправить некоторые моменты и работа будет принята.

Commit:
improved readability of sql queries.
improved the efficiency of delete/update methods.
Add check for the existence of id in the map to TestGetByClient in storage_test.go.

ivannizh commented on internal/storage/storage_test.go:
```diff
@@ -145,6 +146,9 @@ func TestGetByClient(t *testing.T) {
 
 	// check
 	for _, parcel := range storedParcels {
+		if _, ok := parcelMap[parcel.Number]; !ok {
```
Так покрасивше будет
```go
expectedParcel, ok := parcelMap[parcel.Number]
require.True(t, ok)
```

ivannizh reviewed (approved):
Отлично, принято!
